Engineering/Mobile/ios_bluetooth_low_energy/ios_bluetooth_low_energy.md

```diff
@@ -0,0 +1,376 @@
+
+Note 2: indication is a notification with response - Peripheral notifies, Central confirms that notification received.
+
+## IN BLEProofPeripheral SOURCE
+
+* Build BLE Service
+* 
+```
+private func buildBLEService() -> CBMutableService {
+
+        // create characteristics
+        let charForRead = CBMutableCharacteristic(type: uuidCharForRead,
+                                                  properties: .read,
+                                                  value: nil,
+                                                  permissions: .readable)
+        let charForWrite = CBMutableCharacteristic(type: uuidCharForWrite,
+                                                   properties: .write,
+                                                   value: nil,
+                                                   permissions: .writeable)
+        let charForIndicate = CBMutableCharacteristic(type: uuidCharForIndicate,
+                                                      properties: .indicate,
+                                                      value: nil,
+                                                      permissions: .readable)
+        self.charForIndicate = charForIndicate
+
+        // create service
+        let service = CBMutableService(type: uuidService, primary: true)
+        service.characteristics = [charForRead, charForWrite, charForIndicate]
+        return service
+    }
+```
+
+* Handle data reviece from BLE Central
+
+```
+ private func handleDataResponse(_ dataResponse: Data) {
+        var textJson = String(data: dataResponse, encoding: .utf8) ?? ""
+        if let indexRemove = textJson.firstIndex(where: { $0 == "<" }) {
+            textJson.remove(at: indexRemove)
+        }
+        if let indexRemove = textJson.firstIndex(where: { $0 == ">" }) {
+            textJson.remove(at: indexRemove)
+        }
+        convertToJson(with: textJson)
+    }
+    
+    private func convertToJson(with text: String) {
+        guard let data = text.data(using: .utf8),
+              let json = try? JSONSerialization.jsonObject(with: data) as? [String: Any],
+              let pass = json["password"] as? String
+        else {
+            sendNotify(isResponseSuccess: false)
+            return
+        }
+        sendNotify(isResponseSuccess: pass == "1")
+    }
+    
+    private func sendNotify(isResponseSuccess: Bool) {
+        let textSuccess = "<{\"action\":\"bleconfig\",\"result\":\"ok\"}>"
+        let textFail = "<{\"action\":\"bleconfig\",\"result\":\"ng\"}>"
+        bleSendIndication(isResponseSuccess ? textSuccess : textFail)
+    }
+```
+
+* Function didReceiveWrite will receive data from BLE Central
+
+```
+    func peripheralManager(_ peripheral: CBPeripheralManager, didReceiveWrite requests: [CBATTRequest]) {
```

Comment:
Same here, can you remove the leading space and specify the language for the block?